plugins/webp-uploads/settings.php

```diff
@@ -12,11 +12,26 @@
 }
 
 /**
- * Registers setting for generating both JPEG and WebP versions for image uploads.
+ * Registers setting for generating JPEG in addition to the selected modern format for image uploads.
  *
  * @since 1.0.0
+ * @since n.e.x.t The setting was made more general to coveroutputting JPEG as a secondary type. The "webp" option naming
+ *        was left unchanged for backward compatibility.
+ * @since n.e.x.t The `perflab_generate_avif_and_webp` was added to enable selecting between AVIF and WebP output.
  */
 function webp_uploads_register_media_settings_field() {
+	if ( wp_image_editor_supports( array( 'mime_type' => 'image/avif' ) ) ) {
+		register_setting(
+			'media',
+			'perflab_generate_avif_and_webp',
```

Comment:
Would this setting name more sense as `perflab_modern_image_format`? Then it could go beyond AVIF in the future?

Reply:
Good point, I already dislike having "webp" everywhere so no reason to repeat that. Will update.

Reply:
updated